### PR TITLE
Fix worker HPA to target new worker groups

### PR DESCRIPTION
**Pull request by frankh**
Also fix matchLabel clash between different worker deployments

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: charts/posthog/templates/worker-hpa.yaml [hunk 1]
```diff
@@ -1,14 +1,15 @@
 {{- if and .Values.worker.enabled .Values.worker.hpa.enabled -}}
+{{- range .Values.worker.consumers }}
 apiVersion: autoscaling/v2
 kind: HorizontalPodAutoscaler
 metadata:
-  name: {{ template "posthog.fullname" . }}-worker
+  name: {{ template "posthog.fullname" . }}-{{ .name }}-worker
   labels: {{- include "_snippet-metadata-labels-common" . | nindent 4 }}
```
Comment (webjunkie) [inline charts/posthog/templates/worker-hpa.yaml:7]:
```suggestion
  labels: {{- include "_snippet-metadata-labels-common" $ | nindent 4 }}
```

**Review by benjackwhite — APPROVED** [review]

File: charts/posthog/templates/worker-hpa.yaml [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 spec:
   scaleTargetRef:
     kind: Deployment
     apiVersion: apps/v1
-    name: {{ template "posthog.fullname" . }}-worker
+    name: {{ template "posthog.fullname" . }}-{{ .name }}-worker
   minReplicas: {{ .Values.worker.hpa.minpods }}
   maxReplicas: {{ .Values.worker.hpa.maxpods }}
```
Comment (webjunkie) [inline charts/posthog/templates/worker-hpa.yaml:14]:
```suggestion
    name: {{ template "posthog.fullname" $ }}-{{ .name }}-worker
  minReplicas: {{ $.Values.worker.hpa.minpods }}
  maxReplicas: {{ $.Values.worker.hpa.maxpods }}
```

File: charts/posthog/templates/worker-hpa.yaml [hunk 3]
```diff
@@ -23,3 +24,4 @@ spec:
   behavior: 
     {{ toYaml .Values.worker.hpa.behavior | nindent 4 }}
```
Comment (webjunkie) [inline charts/posthog/templates/worker-hpa.yaml:25]:
```suggestion
    {{ toYaml $.Values.worker.hpa.behavior | nindent 4 }}
```

**Review by webjunkie — COMMENTED** [review]
I think these references need to be adjusted.

**Pull request merged**